### DISCUSSION TITLE
CeleryExecutor have all worker logs to stdout

### DISCUSSION THREAD

**Discussion by ricardo-xgeeks** [apache/airflow#25450] [General]
I've noticed that Airflow Workers have some logs being printed to stdout and others going straight to file. As we have a datadog in our setup picking up everything from the pods stdout is there a way to have everything being printed to stdout as we need to process those logs?

With KubernetesExecutor there's a straightforward way to override the pod template for the dynamic pods for injecting custom loggers for example, but I can't seem to find a similar way to do it using CeleryExecutor.

Is there any suggestion as to how to achieve this?

**Comment by potiuk** [comment]
Can you please share examples of what logs you see in stdout that you can't see in the file or the other way round?

**Reply by ricardo-xgeeks** [reply]
@potiuk Basically I believe that the ones being written to file are the task logs and the ones being written to stdout might be from the scheduler.
We would like to have every log hitting stdout in order for Datadog to be able to capture them.

**Reply by potiuk** [reply]
It's better to integrate logging with python logging framework https://airflow.apache.org/docs/apache-airflow/stable/logging-monitoring/logging-tasks.html?highlight=logging#advanced-configuration - capturing stdout misses a lot of important information - line of code it is logged from, logging level, logging parameters. It's much better to use Python loggers to capture data and this is what is recommended by Airflow. 

Do you have a reason that you cannot do that?

**Reply by ricardo-xgeeks** [reply]
Yeah we actually have a custom python logger doing exactly that, we were able to inject it in the celery workers and the logs are being printed to stdout now as we want them to. We've noticed, however, that through Airflow's UI we can't seem to see or download the logs and we're trying to figure out what in our changes might have caused this behavior as it seems that some needed airflow configs have been unset for some reason (like the gunicorn port that the webserver uses to fetch the logs from).

**Reply by potiuk** [reply]
so? What youinveistation shown? Did you follow the advanced task logging and APPENDED handlers rather than replaced them?

**Reply by potiuk** [reply]
Python logging allows to have multiple handlers at every level. So you simply need to add your handler on top of those that are already there when you follow the "advanced logging". But you need to make sure you do not remove the original handlers.

**Reply by ricardo-xgeeks** [reply]
It seems to me that the logger is working properly but right now we're getting a 403 Forbidden from the airflow ui when we check the logs. I'm not getting why this is happening as we didn't mess with the webserver secret key.

**Reply by potiuk** [reply]
Maybe you have different time on two machines. This is often a problem when you have no time synchronized (and if you look at logs of the webserver and if you are on one of the newest versions of Airlfow - which you should be -  it will tell you exactly what is wrong.

**Reply by potiuk** [reply]
Have you actually checked the webserver logs already ??

**Reply by ricardo-xgeeks** [reply]
Yes, I'm checking the webserver logs and I can see a couple of AirflowContextDeprecationWarning entries, but other than that everything looks fine.

**Reply by potiuk** [reply]
Did you upgrade to latest version? Did you check the time on both machines ?

BTW, Without any indication or logs I won't be able to help anyway - so you will need to get deeper to get some evidences.
